### DISCUSSION TITLE
I would like to customize a button in my own VSCode extension, placing it in the same location as the built-in 'Start Debugging' button (Sidebar Toolbar)."

### DISCUSSION THREAD

**Discussion by zivgit** [microsoft/vscode-discussions#1700] [Extension Development QnA] [answered]
I would like to customize a button in my own VSCode extension, placing it in the same location as the built-in 'Start Debugging' button (Sidebar Toolbar)."

<img width="700" alt="sidebar-toolbar-actions" src="https://github.com/user-attachments/assets/8ca5ea88-252a-4644-8f6e-97bcb5fcf49c">

**Comment by robconery** [comment]
It would be super helpful if you could add some detail and ask a specific question that can be answered directly. What is the exact problem?

**Comment by alefragnani** [comment, marked as answer]
Hi @zivgit ,

There is no API available to add buttons on that location. You can only add buttons to the _View’s_ title location (the place that collapse/expand), not the Panel itself.

The print-screen you added, from the documentation (https://code.visualstudio.com/api/ux-guidelines/sidebars#sidebar-toolbars ) is a _special behavior_ that applies to single view Activity Bars .

> However, if only one View is used, the Sidebar will automatically consolidate the UI to use the Sidebar Toolbar to render all of the actions specific to that View. In place of the ... button, the two actions associated with the 'Notes' View are rendered in its place: 

Hope this helps
